core/drivers/ls_gpio.c

```diff
@@ -0,0 +1,213 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright 2021 NXP
+ *
+ * GPIO Driver for GPIO Controller
+ *
+ */
+
+#include <assert.h>
+#include <drivers/ls_gpio.h>
+#include <io.h>
+#include <kernel/boot.h>
+#include <kernel/dt.h>
+#include <libfdt.h>
+#include <mm/core_memprot.h>
+
+#define GPIO_CTRL_NUM  4
+
+static const char *gpio_controller_map[GPIO_CTRL_NUM] = {
```

Comment:
`static const char * const gpio_controller_map[]`,
With this we can remove `GPIO_CTRL_NUM `also